tests/components/librehardwaremonitor/test_sensor.py

```diff
@@ -0,0 +1,132 @@
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_sensors_are_created(
+    hass: HomeAssistant,
+    mock_lhm_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test sensors are created."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    initial_states = hass.states.async_all()
+    assert initial_states == snapshot(name="valid_sensor_data")
+
+    mock_lhm_client.get_data.side_effect = LibreHardwareMonitorConnectionError()
+
+    freezer.tick(timedelta(DEFAULT_SCAN_INTERVAL))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    unavailable_states = hass.states.async_all()
+    assert all(state.state == STATE_UNAVAILABLE for state in unavailable_states)
+
+    mock_lhm_client.get_data.side_effect = None
+
+    freezer.tick(timedelta(DEFAULT_SCAN_INTERVAL))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    recovered_states = hass.states.async_all()
+    assert recovered_states == snapshot(name="valid_sensor_data")
+
+
+async def test_sensors_are_updated(
+    hass: HomeAssistant,
+    mock_lhm_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test sensors are updated."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_id = "sensor.amd_ryzen_7_7800x3d_package_temperature"
+
+    state = hass.states.get(entity_id)
+
+    assert state
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "39.4"
+
+    updated_data = copy.deepcopy(mock_lhm_client.get_data.return_value)
+    updated_data.sensor_data["amdcpu-0-temperature-3"].value = "42,1"
+    mock_lhm_client.get_data.return_value = updated_data
+
+    freezer.tick(timedelta(DEFAULT_SCAN_INTERVAL))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity_id)
+
+    assert state
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "42.1"
+
+    mock_lhm_client.reset_mock()
+
+
+async def test_sensor_state_is_unknown_when_lhm_indicates_missing_value(
+    hass: HomeAssistant, mock_lhm_client: AsyncMock, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test sensor state is unknown when no value is present."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_id = "sensor.msi_mag_b650m_mortar_wifi_ms_7d76_system_fan_1_fan"
+
+    state = hass.states.get(entity_id)
+
+    assert state
+    assert state.state == STATE_UNKNOWN
```

Comment:
isnt this already tested by the snapshot test?

Reply:
You're right, I removed it.